File: InsertionSort/PredefinedInputPython/InsertionSort.py
# create an array of predefined values

# create array


def createArray():
    arr = [9, 8, 7, 6, 5, 4, 3, 2, 1]
    return arr

# print array


def printArray(arr):
    for i in range(len(arr)):
        print(arr[i])

# insertion sort function


def insertionSort(arr):
    for i in range(len(arr)):
        key = arr[i]
        j = i - 1

        while j >= 0 and key < arr[j]:
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = key


arr = createArray()
insertionSort(arr)
printArray(arr)

File: InsertionSort/InsertionSortCS/InsertionSort.cs
using System;

namespace InsertionSortCSharp
{
    class InsertionSort
    {
        public static int[] createArray(int[] array, int size){

            for(int i = size; i > 0; i--){
                array[i-1] = i;
            }
            return array;
        }

        public static void printArray(int[] array, int size){
            for(int i = 0; i < size; i++){
                Console.WriteLine(array[i]);
            }
        }

        public static int[] sort(int[] array, int size){
            int i, j ,key;

            for(i = 1; i < size; i++){
                key = array[i];
                j = i -1;

                while( j>= 0 && array[j] > key ){
                    array[j+1] = array[j];
                    j = j - 1;
                }
                array[j+1] = key;
            }
            return array;
        }
        static void Main(string[] args)
        {
            int size = 100;
            int[] array = new int[size];

            array = createArray(array, size);
            array = sort(array,size);
            printArray(array,size);

        }
    }
}
